### PR TITLE
Update Bing Maps API type definition from V7 to V8 

### PR DESCRIPTION
**Pull request by jamesmcroft**
This refers to the changes asked for in the closed #12683 PR as requested by @andy-ms to move the code into the types-2.0 branch.

### CODE REVIEW

File: bingmaps/v7/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "target": "es6",
+        "noImplicitAny": true,
+        "strictNullChecks": false,
```
Comment (andy-ms) [inline bingmaps/v7/tsconfig.json]: Could this be enabled?

File: bingmaps/bingmaps-tests.ts [hunk 2]
```diff
@@ -1,24 +1,23 @@
...
 /// <reference path="Microsoft.Maps.Search.d.ts"/>
-/// <reference path="Microsoft.Maps.Themes.BingTheme.d.ts"/>
+/// <reference path="Microsoft.Maps.SpatialDataService.d.ts"/>
+/// <reference path="Microsoft.Maps.SpatialMath.d.ts"/>
 /// <reference path="Microsoft.Maps.Traffic.d.ts"/>
-/// <reference path="Microsoft.Maps.VenueMaps.d.ts"/>
+/// <reference path="Microsoft.Maps.WellKnownText.d.ts"/>
 
-namespace BingMapsTests {
+namespace BingMapsV8Tests {
```
Comment (andy-ms) [inline bingmaps/bingmaps-tests.ts]: Namespace/class wrappers don't accomplish anything.

**Review by ghost — COMMENTED** [review]
How are the "modules" (such as autosuggest) used? Right now people will have to explicitly reference files such as `node_modules/@types/bingmaps/Microsoft.Maps.Autosuggest.d.ts` to get those types. Could these be added as references in `index.d.ts` so they are pulled in automatically?

On another note, this might be a good time to add a `tslint.json` containing `{ "extends": "../tslint.json" }` (see [new package instructions](https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package)).

**Comment by mhegazy** [comment]
@jamesmcroft can you update the PR with the changes @andy-ms suggested?

**Comment by andy-ms** [comment]
@jamesmcroft If you still want to do this, please make a new PR targeting the `master` branch.

**Pull request closed without merging**